fast_forward/interaction_fit.py

```diff
@@ -8,203 +8,316 @@
-                                       )
         else:
-            pars = mod.guess(y, x=x)
-
-        fit_result = mod.fit(y, pars, x=x)
-
-        center = fit_result.params["center"].value
-        sigma = fit_result.params["sigma"].value
-
-        func_dict = {'bonds': _bonds_fitter,
-                     'constraints': _bonds_fitter,
-                     'angles': _angles_fitter,
+            center = np.round(np.degrees(gaussian_result.params['center']), self.precision)
+            fc = np.round((self.kb * self.temperature) / ((gaussian_result.params['sigma']) ** 2), self.precision)
+            self.fit_parameters['dihedrals'][group_name] = {'data': data,
+                                                            'distribution_params': [center, fc],
+                                                            'fit_params': {'amp':gaussian_result.params['amplitude'],
+                                                                           'center':gaussian_result.params['center'],
+                                                                           'sigma': gaussian_result.params['sigma']}}
+
+    def fit_to_gmx(self, inter_type, group_name, atoms):
+
+        if inter_type == 'bonds':
+
+            parameters = self.fit_parameters['bonds'][group_name]['distribution_params']
+            center, sigma = parameters
+
+            if sigma < self.constraint_converter:
+                self.interactions_dict['bonds'].append(Interaction(atoms=atoms[0],
+                                                                   parameters=[1, center, sigma],
+                                                                   meta={"comment": group_name}))
+            else:
+                self.interactions_dict['bonds'].append(Interaction(atoms=atoms[0],
+                                                                   parameters=[1, center, 10000],
+                                                                   meta={"ifndef": "FLEXIBLE", "comment": group_name}))
+                self.interactions_dict['constraints'].append(Interaction(atoms=atoms[0],
+                                                                         parameters=[1, center],
+                                                                         meta={"ifdef": "FLEXIBLE",
+                                                                               "comment": group_name}, ))
+        elif inter_type == 'angles':
+
+            parameters = self.fit_parameters['angles'][group_name]['distribution_params']
+            center, sigma = parameters
+
+            # empirically derived. if sigma too big, angles get very unstable.
+            if sigma > 150:
+                sigma = 150
+
+            # empirically derived. For theta_0 > 160, significant ptl energy for type 10 at equilibrium, so enforce type 1.
+            if float(center) < 160:
+                func_type_out = 10
+            else:
+                func_type_out = 1
```

Comment:
This makes sense but also doesn't. Ideally we should check if there are dihedrals covering these angles and log a warning